### PR TITLE
add sample cors test

### PR DESCRIPTION
**Pull request by kevinkreiser**
this is an example that we can expand and duplicate to get the shirtcirtuiters cors functionality merged

### CODE REVIEW

**Comment by kevinkreiser** [comment]
@ldsmoreira i'll get this merged, then we can merge that into your branch and you can expand and duplicate it to test all the functionality that you've added over in #118

File: test/http.cpp [hunk 1]
```diff
@@ -244,6 +252,33 @@ void test_response_parsing() {
...
+  request.version = "HTTP/1.1";
+  auto req_str = request.to_string();
+
+  // send the request
+  http_request_t request_state;
+  server.enqueue("", req_str, request_state);
+
+  // we dont have preflight support implemented yet so we expect nothing to happen here
+  if (server.last_responses.size() != 0)
+    throw std::logic_error("Cors preflight is not implemented but we got a response");
```
Comment (kevinkreiser) [inline test/http.cpp:279]: @ldsmoreira once we get this merged to master and then merged into your branch, you can change this bit here to check that the shortcircuiter replied properly here

File: test/http.cpp [hunk 2]
```diff
@@ -244,6 +252,33 @@ void test_response_parsing() {
     throw std::runtime_error("Response parsing failed");
 }
 
+void test_cors_preflight() {
+  // a server who lets us snoop on what its doing
+  zmq::context_t context;
+  testable_http_server_t server(context, "ipc:///tmp/test_http_server",
```
Comment (kevinkreiser) [inline test/http.cpp:258]: you'll probably also need to configuration your shortcircuiter here as well

**Comment by ldsmoreira** [comment]
Hello @kevinkreiser , what do you think about merging it directly to #118 ? So i can keep working today!
Let me know if i can do that!

**Comment by ldsmoreira** [comment]
Hello @kevinkreiser !! Nice!!
When you get it merged i will start making the tests as soon as possible!

**Comment by kevinkreiser** [comment]
@ldsmoreira it seems that the CI has been updated and it was causes any mac builds to fail. ive opened another pr and merged that, which fixes this (for the tiem being). i've merged it into this branch now and thsi shoudl finally pass CI and i can merge this and unblock you. apologies!

**Pull request merged**